backend/main-service/src/main/java/kkakka/mainservice/coupon/application/DiscountService.java

```diff
@@ -0,0 +1,108 @@
+                product.changeDiscount(discountRequestDto.getDiscount());
+                productRepository.save(product);
+                discountRepository.save(discount);
+                return discount.getId();
+            }
+
+            if (DiscountType.CATEGORY_DISCOUNT.equals(discountRequestDto.getDiscountType())) {
+                Category category = getCategory(discountRequestDto);
+                Discount discount = Discount.create(
+                    category,
+                    discountRequestDto.getName(),
+                    discountRequestDto.getDiscount(),
+                    discountRequestDto.getStartedAt(),
+                    discountRequestDto.getExpiredAt());
+                List<Product> products = categoryRepository.findProductsByCategoryId(category.getId());
+                for (Product product : products) {
+                    product.changeDiscount(discountRequestDto.getDiscount());
+                    productRepository.save(product);
+                }
+                discountRepository.save(discount);
+                return discount.getId();
+            }
+        }
+        throw new KkaKkaException();
+    }
+
+    private Product getProduct(DiscountRequestDto discountRequestDto) {
+        return productRepository.findById(discountRequestDto.getProductId())
+            .orElseThrow(IllegalArgumentException::new);
+    }
+
+    private Category getCategory(DiscountRequestDto discountRequestDto) {
+        return categoryRepository.findById(discountRequestDto.getCategoryId())
+            .orElseThrow(IllegalArgumentException::new);
+    }
+
+    /* 할인 삭제 */
+    @Transactional
+    public void deleteDiscount(Long discountId) {
+        Discount discount = discountRepository.findById(discountId)
+            .orElseThrow(KkaKkaException::new);
+        if (DiscountType.PRODUCT_DISCOUNT.equals(discount.getDiscountType())) {
+            Product product = productRepository.findById(discount.getProductId())
+                .orElseThrow(KkaKkaException::new);
+            discount.deleteDiscount();
+            product.deleteDiscount();
+            discountRepository.save(discount);
+            productRepository.save(product);
+        }
+        if (DiscountType.CATEGORY_DISCOUNT.equals(discount.getDiscountType())) {
+            List<Product> products = categoryRepository.findProductsByCategoryId(
+                discount.getCategoryId());
+            for (Product product : products) {
+                product.deleteDiscount();
+                productRepository.save(product);
+            }
+            discount.deleteDiscount();
+            discountRepository.save(discount);
+        }
+    }
+
+    /* 할인 조회 */
+    public List<DiscountResponseDto> showAllDiscounts() {
+        List<Discount> discounts = discountRepository.findAll();
+        return discounts.stream().map(discount -> DiscountResponseDto.create(discount)).collect(
+            Collectors.toList());
```

Comment:
```suggestion
        return discounts.stream()
                .map(discount -> DiscountResponseDto.create(discount))
                .collect(Collectors.toList());
```
이런 의미있는 메서드 체이닝으로 이루어졌을 경우에 개행을 어떻게 하느냐에 따라 가독성이 많이 달라진다고 생각합니다ㅎㅎ

Reply:
자동정렬했더니 저렇게 됐네요 😅